### PR TITLE
German translation

### PR DESCRIPTION
**Pull request by pikim**

### CODE REVIEW

**Comment by pikim** [comment]
Error in translation. Update follows. Sorry!

**Comment by pikim** [comment]
Github already considered the fix.

**Review by pmoreno-rodriguez — COMMENTED** [review]
For Future 1.0.3 version

**Review by pmoreno-rodriguez — COMMENTED** [review]
For Future 1.0.3 version

**Review by pmoreno-rodriguez — APPROVED** [review]
Thanks for your contribution. All changes will be included in the new version (1.0.3)

**Comment by pikim** [comment]
OK, thanks for merging. I wanted to mention that some commits lead to differences to the original future imperfect theme. These are especially:
[Disable (partial) logo hiding](https://github.com/pmoreno-rodriguez/grav-theme-future2021/pull/7/commits/992b43839fb78e4b3e950dec1ff6c12c495c7cca)
[Optimize distances in sidebarmenu](https://github.com/pmoreno-rodriguez/grav-theme-future2021/pull/7/commits/ffcc69af2a361262851f8eb901bcd3a4e27ff9b0)

So, if someone doesn't like those, one knows what to revert.

**Comment by pmoreno-rodriguez** [comment]
Have you some website running with this theme and your contributions?.

**Comment by pikim** [comment]
I'm preparing one, but up to now it only runs on my local machine. But a draft will be online soon. I can share the link, then.

[Harmonize and modify twigs](https://github.com/pmoreno-rodriguez/grav-theme-future2021/pull/7/commits/9dc080bed82650c2f23403651af0f1d094fa30cd)
breaks some pages as I removed some features I didn't want and accidentally committed them here. I'm working on a fix. You shouldn't release a new version at the moment. I'll inform you as soon as it's fixed. Sorry for that!

**Comment by pmoreno-rodriguez** [comment]
I have been testing your contributions and have found some problems:

- The event and calendar templates need some plugins to work and they don't work well.
- I think the #intro .logo:before class should go back to the original look, to preserve the original design of the Future Imperfect theme
- The link to the author's posts (when you click on the author's avatar image) doesn't work, nor do the tag links, below each post. I think the base_url in the blog template is wrong, although they worked in the previous version. (I'm working in it)
- The margins between the last blog posts in the sidebar, should be those of the original theme, now they appear too close.
- The link of the login button must point to the login form, of the plugin form, and not to the administration page (I'm working on it)
- Events, Calendar, Downloads, Guestbook templates do not have associated blueprints and need third-party plugins, so it is difficult for a user to know what to do when adding these templates (remember that not all Grav users are programmers).
- I don't understand why you created a Gallery template, if the theme already has a portfolio that makes use of a Javascript library and doesn't need a 3rd party plugin. It also does not have any associated blueprint.

In short, I think I should have reviewed your contributions better to avoid correcting now after making the committees. I'm still not very familiar with Github and I think it would be better to commit to the development branch first. Going forward, I will follow this recommendation.

**Comment by pikim** [comment]
In general I like the theme, but there are some details I don't like and therefore I made a couple of changes to adjust those things. And initially I didn't want to pull request everything and I still don't know why all following commits also became part of the request. How you can revert all commits until a specific version is described [here](https://stackoverflow.com/a/21718540). I'd suggest that you revert all my changes and I'll create new pull requests with smaller chunks that don't change the appearance.

As far as I understood the gallery you use must be licensed and therefore I went for the free unitegallery.

Regarding dependencies: does your portfolio-twig work right out of the box without further plugin installation?

For clarification: calendar and events require the events-plugin, guestbook requires the guestbook-plugin, downloads doesn't require anything.

I'm preparing a online draft site at the moment and I'll share a link as soon as possible.

**Comment by pmoreno-rodriguez** [comment]
The portfolio template uses the [Glightbox library](https://github.com/biati-digital/glightbox), which is licensed under the MIT license and is freely distributed for personal and commercial use. The portfolio template uses this library and starts from base.html.twig.

You can take a look at the base.html.twig, portfolio.yaml, and portfolio.html.twig files to see how it works.

**Comment by pikim** [comment]
OK, I see. Then I didn't get that correctly.

What's your plan regarding the additional twigs? In the meantime I learnt that theme inheritance is the right way and therefore you can remove my twigs again. It's your theme and therefore you decide which features it supports. I'll make a customization which inherits from your base version and place any changes and extensions there.

The easiest for you would be to revert all my changes and merge only changes that make sense. It would be absolutely ok for me. You needn't clean up my mess.

Sorry for the inconvenience I caused!

**Comment by pmoreno-rodriguez** [comment]
Relax, do not worry. I'll roll back those changes that break the functionality or look of the theme, and leave the ones that work.

**Comment by pikim** [comment]
OK, nice to hear that.

You should rename css/custom.css as this is more or less reserved for theme inheritance as you can see [here](https://getgrav.org/blog/theme-development-with-inheritance). At least it would make much sense to do so to avoid others falling over that.

**Comment by pikim** [comment]
I'm still wanting to support you with cleaning up my mess but respect your future plans.

After comparing the latest revision with the version before I started to mess things:
https://github.com/pmoreno-rodriguez/grav-theme-future2021/compare/83fae7ea06d80fe99d034375f2a8ed93377590e7...develop

I'd say the following files also have to be reverted to get back to the initial style and functionality:
assets/css/main.css
assets/sass/components/_post.scss
assets/sass/components/_posts.scss
assets/sass/components/_sidebarmenu.scss
templates/partials/sidebar_right.html.twig

The following files were added by me and can be safely removed without breaking any functionality:
templates/calendar.html.twig
templates/downloads.html.twig
templates/event.html.twig
templates/events.html.twig
templates/gallery.html.twig
templates/guestbook.html.twig
templates/partials/download_item.html.twig
templates/partials/event_item.html.twig
templates/partials/guestbook_item.html.twig

**Comment by pmoreno-rodriguez** [comment]
Hello. I'm reviewing all changes so that both your input and my future plans for the theme can be in the next version. Thanks for your help

**Comment by pmoreno-rodriguez** [comment]
Hello @pikim.
I am going to release a new version of the Future2021 theme (view [develop branch](https://github.com/pmoreno-rodriguez/grav-theme-future2021)), with new features and some minor bug fixes.
The language file has new items translated into English and Spanish, but some German translations are missing.
Would you like to make those translations?
Thank you so much.

**Comment by pikim** [comment]
Hi @pmoreno-rodriguez 
I'll have a look. How do you plan to handle the English texts? There are many words with capital first letters which is not really correct, e.g. Custom **L**ogo?

**Comment by pmoreno-rodriguez** [comment]
@pikim, your suggestions will be welcome. Possibly, I have much more problems with English than you, so your help will be important to improve the subject.

**Comment by pikim** [comment]
@pmoreno-rodriguez
OK. Hmm, the theme doesn't appear in my themes section. I'll try to find out why.

**Comment by pmoreno-rodriguez** [comment]
> Hi @pmoreno-rodriguez I'll have a look. How do you plan to handle the English texts? There are many words with capital first letters which is not really correct, e.g. Custom **L**ogo?

Have you seen the Quark theme template? It has words with first capital letters (Custom Logo, etc). However, if you think it's important to fix some linguistic errors, we'll do it.

**Comment by pikim** [comment]
@pmoreno-rodriguez 
It's not important, but it's inconsistent because it's not the same throughout the whole theme. Sometimes I'm pedantic regarding such topics ;-)

Just out of curiosity: is the frequent nesting of `fields` in `default.yaml` (and once in `blog.yaml`) intentionally and necessary?

I think I can finish it tomorrow in the evening.

I'll have to think about how I'll handle updated versions in the future. I'm still on v1.0.2 as I must adjust my customization to the newer version first. As some styling has changed I must check every page.

**Comment by pikim** [comment]
@pmoreno-rodriguez 
Could you provide a screenshot of the output of default.yaml? I have the impression that this doesn't work here.

**Comment by pmoreno-rodriguez** [comment]
@pikim, Do you mean the .yaml code? It is in the [development branch](https://github.com/pmoreno-rodriguez/grav-theme-future2021/blob/develop/blueprints/default.yaml).

**Comment by pikim** [comment]
@pmoreno-rodriguez 
I meant how it looks in the admin interface then?

I would have expected that it adds some settings to the configuration of 'default' sites. But it can't see any.

**Comment by pmoreno-rodriguez** [comment]
@pikim,

Here you have some screenshots about default template in admin:

**Set Show Sidebar option for all pages based in default**

default_template2

**Set help to some body classes supported by Future Imperfect theme**

default_template3

**Set subtitle and primaryImage**

default_template


> I would have expected that it adds some settings to the configuration of 'default' sites. But it can't see any.

I don't quite understand what you mean by this.

**Comment by pikim** [comment]
> I don't quite understand what you mean by this.

The images helped me. I had some issues due to an outdated config and therefore wasn't able to see the options that were added by the theme. But it works now.

**Comment by pikim** [comment]
@pmoreno-rodriguez 
- wouldn't `BLOG_CONTENT_ITEMS_SIDEBAR` make more sense in `blog.yaml` instead of `default.yaml`?
- could the last line in `default.yaml` (`description: 'Available classes in Future2021 Theme:<br />`single`'`) also be translated? I tried it but without success.

**Comment by pmoreno-rodriguez** [comment]
> @pmoreno-rodriguez
> 
> * wouldn't `BLOG_CONTENT_ITEMS_SIDEBAR` make more sense in `blog.yaml` instead of `default.yaml`?
> * could the last line in `default.yaml` (`description: 'Available classes in Future2021 Theme:<br />`single`'`) also be translated? I tried it but without success.

Take a look at the languages and default template file in the development branch to see the translation of the body classes.

On the other hand, the translation of showing sidebar was originally only for the blog, but I decided to extend it to all pages. That's why the translation is still in the blog section

**Comment by pmoreno-rodriguez** [comment]
> * BLOG_CONTENT_ITEMS_SIDEBAR

Hi @pikim.
I have finished reviewing the Future2021 theme source code for the upcoming version 1.0.4. There is only one German translation left, which I don't know how to translate. (you can take a look at the development branch)

Would you like to finish those translations?

**Comment by pikim** [comment]
Hi @pmoreno-rodriguez 
I just finished the translation and merged it already. Could you please have a look on `MORE_TEXT_DEFAULT` which I also added as it didn't exist before?
